command/genesis/predeploy/params.go

```diff
@@ -26,34 +30,48 @@ var (
 	errInvalidAddress          = fmt.Errorf(
 		"the provided predeploy address must be >= %s", predeployAddressMin.String(),
 	)
+	errArtifactPathAndNameMissing = errors.New("neither artifact path nor artifact name was provided")
 
 	predeployAddressMin = types.StringToAddress("01100")
 	params              = &predeployParams{}
 )
 
 type predeployParams struct {
-	addressRaw  string
-	genesisPath string
+	addressRaw      string
+	genesisPath     string
+	deployerAddrRaw string
+
+	address       types.Address
+	deployerAddr  types.Address
+	artifactsName string
+	artifactsPath string
 
-	address         types.Address
-	artifactsPath   string
 	constructorArgs []string
 
 	genesisConfig *chain.Chain
+	scArtifact    *contracts.Artifact
 }
 
 func (p *predeployParams) getRequiredFlags() []string {
 	return []string{
 		predeployAddressFlag,
-		artifactsPathFlag,
+		artifactsNameFlag,
 	}
 }
 
-func (p *predeployParams) initRawParams() error {
+func (p *predeployParams) initRawParams() (err error) {
+	if p.artifactsName == "" && p.artifactsPath == "" {
+		return errArtifactPathAndNameMissing
+	}
+
 	if err := p.initPredeployAddress(); err != nil {
 		return err
 	}
```

Comment:
```suggestion
	var err error  
    
    p.address, err = types.IsValidAddr()  
    if err != nil {
        return errInvalidPredeployAddress
```

Also `initPredeployAddress` can be removed in that case.